tests/language-feature/enums/zero-initialize/generic.slang

```diff
@@ -0,0 +1,88 @@
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -xslang -zero-initialize
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-vk -compute -entry computeMain -emit-spirv-directly -allow-glsl -xslang -zero-initialize
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-cpu -compute -entry computeMain -allow-glsl -xslang -zero-initialize
+//TEST(smoke,compute):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-dx12 -use-dxil -compute -entry computeMain -allow-glsl -xslang -zero-initialize -xslang -DDX12
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+__generic<T : __BuiltinArithmeticType>
+bool getValue()
+{
+    T genericOut;
+    return genericOut == T(0);
+}
+__generic<T : __BuiltinArithmeticType, let N : int>
+bool getValueV()
+{
+    typealias gvec = vector<T, N>;
+
+    gvec genericOut;
+    return genericOut == gvec(T(0));
+}
+[numthreads(1, 1, 1)]
+void computeMain(int3 dispatchThreadID: SV_DispatchThreadID)
+{
+// BUF: 1
+    outputBuffer[0] = true
+    && getValue<int>()
+#ifndef DX12
+    && getValue<uint8_t>()
+    && getValue<int8_t>()
+    && getValue<uint16_t>()
+    && getValue<int16_t>()
+#endif
```

Comment:
8bit is not supported by dx12 but 16bit is.
Didn't 16bit work?

Reply:
the types work, I forgotten to set the profile to sm_6_2.